patches/api/0480-Add-preliminary-API-for-working-with-new-mob-AI-syst.patch

```diff
@@ -0,0 +1,236 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Janet Blackquill <uhhadd@gmail.com>
+Date: Fri, 26 Apr 2024 23:45:11 -0400
+Subject: [PATCH] Add preliminary API for working with 'new' mob AI system
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/ai/Activity.java b/src/main/java/io/papermc/paper/entity/ai/Activity.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..0dff7ba2e36c1185453832e28466c82ccb2cd621
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/ai/Activity.java
+@@ -0,0 +1,9 @@
++package io.papermc.paper.entity.ai;
++
++import org.bukkit.Keyed;
++
++/**
++ * An Activity is a key to a list of tasks in a Brain.
++ */
++public interface Activity extends Keyed {
++}
+diff --git a/src/main/java/io/papermc/paper/entity/ai/Brain.java b/src/main/java/io/papermc/paper/entity/ai/Brain.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..d1f170f12e1a3403ff6b392460cb9e2c5b1f5eeb
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/ai/Brain.java
+@@ -0,0 +1,53 @@
++package io.papermc.paper.entity.ai;
++
++import java.util.List;
++import org.bukkit.entity.LivingEntity;
++import org.bukkit.entity.Entity;
++import org.bukkit.Location;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * A brain is the central AI for some modern Minecraft entities such as
++ * villagers and sniffers.
++ */
++public interface Brain<E extends LivingEntity> {
++    /**
++     * If the tasks in the given `activity` can be run, the brain will switch to
++     * the given `activity`.
++     */
++    public void useActivityIfPossible(@NotNull Activity activity);
```

Comment:
This applies to all additions - interfaces do not need methods to specify `public`.

```suggestion
+    void useActivityIfPossible(@NotNull Activity activity);
```

Reply:
fixed